velox/functions/sparksql/aggregates/tests/SumAggregationTest.cpp

```diff
@@ -449,5 +449,16 @@ TEST_F(SumAggregationTest, decimalRangeOverflow) {
       {expected},
       {});
 }
+
+TEST_F(SumAggregationTest, sumFloat) {
+  auto data = makeRowVector({makeFlatVector<float>({2.00, 1.00})});
```

Comment:
Did this test fail before the change?

Reply:
Yes

Reply:
@FelixYBW I wonder what was the failure. I tried running this test on 'main' and it passed.

Reply:
:( @JkSelf Did you do the test? Looks the UT can't detect the type mismatch.

The PR does solved the customer issue but the issue is caused by window function validation.

Reply:
@mbasmanova @FelixYBW 

When running the sum(float) aggregate window function with gluten, the following error occurs:

```
Error Source: USER
Error Code: INVALID_ARGUMENT
Reason: Unexpected return type for window function sum(REAL). Expected REAL. Got DOUBLE.
Retriable: False
```

This is due to a mismatch between the function signatures in Spark and Velox. In Spark, the sum(real) function is expected to return a double type, whereas in Velox, the same function is registered to return a real type, leading to incompatibility. It is hard to reproduce this exception in Velox. I have modified the unit test to trigger an overflow error if the current patch is not applied. Please help to review again. Thanks.